spec.html

```diff
@@ -9401,12 +9441,10 @@ <h1>NewDeclarativeEnvironment ( _E_ )</h1>
       </emu-clause>
 
       <emu-clause id="sec-newobjectenvironment" aoid="NewObjectEnvironment">
-        <h1>NewObjectEnvironment ( _O_, _E_ )</h1>
-        <p>The abstract operation NewObjectEnvironment takes arguments _O_ (an Object) and _E_ (an Environment Record). It performs the following steps when called:</p>
+        <h1>NewObjectEnvironment ( _O_, _W_, _E_ )</h1>
+        <p>The abstract operation NewObjectEnvironment takes arguments _O_ (an Object), _W_ (a Boolean) and _E_ (an Environment Record). It performs the following steps when called:</p>
         <emu-alg>
-          1. Let _env_ be a new object Environment Record containing _O_ as the binding object.
-          1. Set _env_.[[OuterEnv]] to _E_.
-          1. Return _env_.
+          1. Return object Environment Record { [[OuterEnv]]: _E_, [[BindingObject]]: _O_, [[WithEnvironmentFlag]]: _W_ }.
```

Comment:
Nit: the inline record style here doesn't match the style in the other New*Environment methods. I'd prefer they match, probably by having this be the same style as the others: so

```
1. Let _env_ be a new object Environment Record.
1. Set _env_.[[BindingObject]] to _O_.
1. Set _env_.[[WithEnvironmentFlag]] to _W_.
1. Set _env_.[[OuterEnv]] to _E_.
1. Return _env_. 
```

Reply:
I'm on the fence about whether that's an improvement. (Perhaps best would be a direction on issue #2095.) But I'll make the change if that's the editors' consensus.

Reply:
Partly I prefer it because most of the other environment record kinds aren't _just_ records; they have a somewhat magical collection of bindings associated with them as well (hence the "containing no bindings" phrase which appears when creating them). So at least until that's rectified I wouldn't want to format them with this style, regardless of #2095. And then for consistency I would want NewObjectEnvironmentRecord to match its fellows.

Reply:
Done.